### PR TITLE
UI: make a newly added Network the default one

### PR DESCRIPTION
**Pull request by abh1sar**
Checkbox to make a newly added Network the default one in the Instance's Network tab

### Description

This PR adds a checkbox into the Add Network dialogue to make the newly added network the default one in Instance's Network Tab. This will save users from performing an extra step to make the network the default one. 
 

### Types of changes

- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] Enhancement (improves an existing feature and functionality)
- [ ] Cleanup (Code refactoring and cleanup, that may add test cases)
- [ ] build/CI
- [ ] test (unit or integration test code)

#### Feature/Enhancement Scale

- [ ] Major
- [x] Minor

#### Bug Severity

- [ ] BLOCKER
- [ ] Critical
- [ ] Major
- [ ] Minor
- [ ] Trivial

### Screenshots (if appropriate):
Screenshot from 2024-09-19 11-13-08
Screenshot from 2024-09-19 11-13-19
Screenshot from 2024-09-19 11-13-22

### CODE REVIEW

**Comment by abh1sar** [comment]
@blueorangutan ui

File: ui/src/views/network/NicsTab.vue [hunk 1]
```diff
@@ -368,9 +376,26 @@ export default {
...
+                  this.loadingNic = false
+                  this.closeModals()
+                }
+              } catch (error) {
+                this.$notifyError('Failed to fetch NIC data.')
+                this.loadingNic = false
+                this.closeModals()
+              }
+            } else {
+              this.loadingNic = false
```
Comment (weizhouapache) [inline ui/src/views/network/NicsTab.vue:396]: can the following be moved to the line before `errorMessage` ?

```
              this.loadingNic = false
              this.closeModals()
```

Comment (abh1sar) [reply ui/src/views/network/NicsTab.vue]: Made some changes to the code. Before that the active tab was getting reset to Details on Add-Network.
Now the Network tab remains the active tab after adding a network.
Please check if this comment still applies.

**Comment by abh1sar** [comment]
@blueorangutan ui

File: ui/src/views/network/NicsTab.vue (hunk 1 above)
Comment (weizhouapache) [reply ui/src/views/network/NicsTab.vue]: @abh1sar 
the changes in your last commit https://github.com/apache/cloudstack/pull/9708/commits/02be6a2e48e906d5256ab683747d62709bb73a52 is exactly what I meant.

**Review by weizhouapache — APPROVED** [review]
tested ok on qa

**Review by sureshanaparti — APPROVED** [review]
clgtm

**Comment by kiranchavala** [comment]
@blueorangutan test

**Review by rohityadavcloud — APPROVED** [review]

**Review by rajujith — APPROVED** [review]
LGTM

**Comment by rohityadavcloud** [comment]
@JoaoJandre @DaanHoogland some PRs targeting 4.20.1 are merged on main - should they go to 4.20 branch?

**Comment by JoaoJandre** [comment]
@rohityadavcloud Considering that they were not targeting 4.19 (a maintenance branch), I assumed the authors wanted their changes to be made on the main branch directly. I'll fix the milestone. If the authors want their changes to be on the 4.20 branch, I volunteer to backport the PRs (https://github.com/apache/cloudstack/pull/9930, https://github.com/apache/cloudstack/pull/9708, https://github.com/apache/cloudstack/pull/9956).

**Comment by abh1sar** [comment]
@JoaoJandre This is a small change but very useful. It would be great if we can put this to 4.20.1 as well.

**Pull request merged**